index.js

```diff
@@ -1,162 +1,152 @@
-              "pageSrc": window.location.href,
-              "pageTitle": document.title,
-              "mimeType": mimeType,
-              "duration": clamp_duration(target.duration),
-              "detected": true,
-              "tagId": target.tagUUID,
-              thumbnail
-            });
+              'name': name,
+              'src': fixUpRelativeUrl(node.src),
+              'pageSrc': window.location.href,
+              'pageTitle': document.title,
+              'mimeType': mimeType,
+              'duration': getMediaDurationInSeconds(target),
+              'detected': true
+            })
           }
 
           if (node.closest('audio') === target) {
-            tagNode(target);
             sources.push({
-              "name": name,
-              "src": node.src,
-              "pageSrc": window.location.href,
-              "pageTitle": document.title,
-              "mimeType": mimeType,
-              "duration": clamp_duration(target.duration),
-              "detected": true,
-              "tagId": target.tagUUID,
-              thumbnail
-            });
+              'name': name,
+              'src': fixUpRelativeUrl(node.src),
+              'pageSrc': window.location.href,
+              'pageTitle': document.title,
+              'mimeType': mimeType,
+              'duration': getMediaDurationInSeconds(target),
+              'detected': true
+            })
           }
         }
-        
-      });
-      return sources;
+      })
+      return sources
     }
   }
 
-  function getNodeData(node, thumbnail) {
-    return getNodeSource(node, node.src, node.type, thumbnail);
+  function getAllVideoElements () {
+    return document.querySelectorAll('video')
   }
 
-  function getAllVideoElements() {
-    return document.querySelectorAll('video');
+  function getAllAudioElements () {
+    return document.querySelectorAll('audio')
   }
 
-  function getAllAudioElements() {
-    return document.querySelectorAll('audio');
+  function getThumbnail () {
+    const isThumbnailValid = (thumbnail) => { return thumbnail && thumbnail !== '' }
+
+    let thumbnail = document.querySelector('meta[property="og:image"]')?.content
+    return fixUpRelativeUrl(thumbnail)
   }
 
-  function getOGTagImage() {
-    return document.querySelector('meta[property="og:image"]')?.content
+  function getMediaTitle (node) {
+    const isTitleValid = (title) => { return title && title !== '' }
+
+    let title = node.title
+    if (!isTitleValid(title)) { title = document.title }
+
+    return title
+  }
+
+  function getMediaAuthor (node) {
+    // TODO(sko) Get metadata of author
+    return null
   }
 
-  let videoElements = getAllVideoElements() ?? [];
-  let audioElements = getAllAudioElements() ?? [];
-  const thumbnail = getOGTagImage();
+  function getMediaDurationInSeconds (node) {
+    let duration = node.duration
+    const isNan = (value) => { return typeof value === 'number' && Number.isNaN(value) }
```

Comment:
Note: This will return false for everything (including strings/objects/arraYS/booleans) except the literal value `NaN`. Is that what we want?

Optional nit:
```suggestion
    const isNan = (value) => typeof value === 'number' && Number.isNaN(value)
```

Reply:
Yes, I think so. This part is from https://github.com/brave/brave-ios/blob/bb1832c245c0e22d1876a544e2cde4a2356863ef/Client/Frontend/UserContent/UserScripts/Scripts_Dynamic/Scripts/Sandboxed/PlaylistScript.js#L9

Reply:
Yeah, there's also `Number.isFinite` for checking it's not `Infinity` or `NaN` or `nullish`. In addition the `isInfinite` check here is a bit weird - `typeof Infinity === "number"` so it doesn't really add anything to the check...